pytket/pytket/qasm/qasm.py

```diff
@@ -1597,9 +1595,8 @@ def remove_unused_predicate(self, pred_label: int) -> bool:
             string = self.strings.get_string(label)
             if string is None:
                 continue
-            if (
-                _var_appears(pred.dest, string)
-                or label in self.strings.conditions
+            if _var_appears(pred.dest, string) or (
+                label in self.strings.conditions
```

Comment:
Why the new condition here?

Reply:
Not sure what you mean? I think the conditions are all in there already?

Reply:
I was hallucinating, sorry.